compiler/AST/build.cpp

```diff
@@ -2194,7 +2194,8 @@ FnSymbol* buildLinkageFn(Flag externOrExport, Expr* paramCNameExpr) {
   }
 
   // Handle non-trivial param names that need to be resolved
-  if (paramCNameExpr && cname[0] == '\0') {
+  // the check for dtString->symbol avoids this block under chpldoc
+  if (paramCNameExpr && cname[0] == '\0' && dtString->symbol != NULL) {
     DefExpr* argDef = buildArgDefExpr(INTENT_BLANK,
```

Comment:
I think I would prefer this argument to have a special flag or something that chpldoc could key off of, instead of altering the behavior at parse time depending on our compilation mode

Reply:
Something is already (very) altered under chpldoc as dtString->symbol == NULL there, so we get assert failures from buildArgDefExpr (before the main chpldoc traversal even runs).

Reply:
Ah, okay.  That's a good justification.  I know we did some modification so that we don't generate documentation for internal symbols, so it makes sense that is happening (but is less than ideal imo)